src/aks-preview/azext_aks_preview/maintenanceconfiguration.py

```diff
@@ -47,35 +48,19 @@ def aks_maintenanceconfiguration_update_internal(
 ):
     logger.info('resource_group_name: %s, cluster_name: %s, config_name: %s, config_file: %s, weekday: %s, start_hour: %s ', resource_group_name, cluster_name, config_name, config_file, weekday, start_hour)
 
-    config = getMaintenanceConfiguration(cmd, config_file, weekday, start_hour)
+    config = getMaintenanceConfiguration(cmd, client, config_file, weekday, start_hour)
     return client.create_or_update(resource_group_name=resource_group_name, resource_name=cluster_name, config_name=config_name, parameters=config)
 
 
-def _get_maintenance_config(cmd, file_path):
+def _get_maintenance_config(cmd, client, file_path):
```

Comment:
Actually, we don't need to call msrest in cli since SDK would perform serialization and deserialization for us. You could try the following, this should work as it used to be.
```suggestion
def _get_maintenance_config(cmd, file_path):
    maintenance_config = get_file_json(file_path)
    return maintenance_config
```

Reply:
I am not sure whether I got your idea. I checked LinuxOSConfig & MangedClusterHTTPProxyConfig, looks they read the json file, then manually populate the properties into REST model object which is what we want to avoid.

Reply:
I am not sure if you got the right place. Please check the getter functions for `--kubelet-config`, `--linux-os-config` and `--http-proxy-config`.

- [get_kubelet_config](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/decorator.py#L374)
- [get_linux_os_config](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/decorator.py#L414)
- [get_http_proxy_config](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/decorator.py#L454)